fairworkflows/fairworkflow.py

```diff
@@ -86,24 +86,29 @@ def from_function(cls, func: Callable):
                              'use is_fairworkflow decorator to mark it.')
 
     @classmethod
-    def from_noodles_promise(cls, noodles_promise, description: str = None, label: str = None,
-                 is_pplan_plan: bool = True, derived_from=None):
+    def from_noodles_promise(cls, workflow_level_promise, step_level_promise, description: str = None, label: str =
```

Comment:
Perhaps we should add type hints here to ensure noodles promises are being passed?

Reply:
Added :)